packages/mattermost-redux/src/actions/channel_categories.test.js

```diff
@@ -327,6 +327,89 @@ describe('fetchMyCategories', () => {
         expect(state.entities.channelCategories.byId.category2).toEqual(categories[1]);
         expect(state.entities.channelCategories.orderByTeam[teamId]).toEqual(['category1', 'category2']);
     });
+    test('should update collapse state if it s from websocket', async () => {
+        const currentUserId = TestHelper.generateId();
+        const teamId = TestHelper.generateId();
+        const isWebSocket = false;
+
+        const categories = [
+            {
+                id: 'category1',
+                collapsed: false,
+                team_id: teamId,
+            },
+            {
+                id: 'category2',
+                collapsed: true,
+                team_id: teamId,
+            },
+        ];
+
+        const store = configureStore({
+            entities: {
+                users: {
+                    currentUserId,
+                },
+                channelCategories: {
+                    byId: {
+                        category1: {id: 'category1', team_id: teamId, collapsed: true},
+                        category2: {id: 'category2', team_id: teamId, collapsed: false},
+
+                    },
+                },
+            }});
+
+        nock(Client4.getBaseRoute()).
+            get(`/users/${currentUserId}/teams/${teamId}/channels/categories`).
+            reply(200, {categories, order: categories.map((category) => category.id)});
```

Comment:
For these tests to be effective, nock should return the categories with a different `collapsed` value than we already have in the store since that'd cause a state change with the previous code. Could you change these so that the response from the server here has the `categories` set with different values for their `collapsed`?

Reply:
> These are actually unit tests, not end-to-end tests,
Very sorry for the confusion
>  I don't actually know how the E2E tests would work with Gitpod anyway.
I can run the webapp tests locally as it doesn't require the backend to be working however if E2E tests depend on the backend to be working I can not do it without Gitpod.

Reply:
No worries. I was debating on if I should be specific about which type of tests I wanted since either of them would probably work fine here

Reply:
I've just bought a new laptop and I've been super excited to be able to run mattermost server and webapp locally. 
I will pick up where I left off i the rest of PRs in the following days.

Reply:
> For these tests to be effective, nock should return the categories with a different collapsed value than we already have in the store since that'd cause a state change with the previous code.

That's exactly what's happening: 
`   const categories = [
            {
                id: 'category1',
                collapsed: false,
                team_id: teamId,
            },
            {
                id: 'category2',
                collapsed: true,
                team_id: teamId,
            },
        ];
`
this is the values that are being passed to Nock.
then I set up the store  with different values.
`   const categories = [
            {
                id: 'category1',
                collapsed: false,
                team_id: teamId,
            },
            {
                id: 'category2',
                collapsed: true,
                team_id: teamId,
            },
        ];
`
then on the first test I check if the result matches the one I sent to nock (since it's not from a websocket)
the second test I check if the result match the store and **not** the values sent to nock

Am I missing something else?

Reply:
Oh, I can't read apparently. Sorry about that. I thought the same categories were being passed into `configureStore` that were being returned from `nock`. This looks good to me then 😅